File: mobile-engage/src/androidTest/java/com/emarsys/mobileengage/testUtil/SharedPrefsUtils.java
package com.emarsys.mobileengage.testUtil;

import android.content.Context;
import android.support.test.InstrumentationRegistry;

import com.emarsys.mobileengage.storage.Storage;

public class SharedPrefsUtils {
    public static final String MOBILE_ENGAGE_SHARED_PREFS = Storage.SHARED_PREFERENCES_NAMESPACE;

    private SharedPrefsUtils() {
    }

    public static void deleteMobileEngageSharedPrefs() {
        InstrumentationRegistry
                .getTargetContext()
                .getSharedPreferences(MOBILE_ENGAGE_SHARED_PREFS, Context.MODE_PRIVATE)
                .edit()
                .clear()
                .commit();
    }

}

File: testUtils/src/main/java/com/emarsys/testUtil/SharedPrefsUtils.kt
package com.emarsys.testUtil

import android.content.Context

object SharedPrefsUtils {

    @JvmStatic
    @JvmOverloads
    fun clearSharedPrefs(namespace: String, mode: Int = Context.MODE_PRIVATE) =
            InstrumentationRegistry
                    .getTargetContext()
                    .getSharedPreferences(namespace, mode)
                    .edit()
                    .clear()
                    .commit()

}